src/anyio/_core/_sockets.py

```diff
@@ -591,8 +601,11 @@ def getnameinfo(sockaddr: IPSockAddrType, flags: int = 0) -> Awaitable[tuple[str
     return get_async_backend().getnameinfo(sockaddr, flags)
 
 
+@deprecated("This function is deprecated; use `wait_readable` instead", stacklevel=2)
```

Comment:
Wait you need to set the stack level here? That can't be right

Reply:
Default value is `1`, would you prefer that?

Reply:
Ah no you don't, the stack level is already set to +1 https://github.com/python/cpython/blob/3.13/Lib/warnings.py

```suggestion
@deprecated("This function is deprecated; use `wait_readable` instead")
```

Reply:
The default is 1, but that then calls warn with 2. If we set 2 then warn gets called with 3